src/query/functions/src/scalars/string.rs

```diff
@@ -757,6 +758,73 @@ pub fn register(registry: &mut FunctionRegistry) {
             }
         }),
     );
+
+    registry
+        .register_passthrough_nullable_2_arg::<StringType, StringType, ArrayType<StringType>, _, _>(
+            "split",
+            |_, _, _| FunctionDomain::Full,
+            vectorize_with_builder_2_arg::<StringType, StringType, ArrayType<StringType>>(
+                |str, sep, output, ctx| match String::from_utf8(str.to_vec()) {
+                    Ok(s) => match String::from_utf8(sep.to_vec()) {
+                        Ok(sep) => {
+                            let res: Vec<&str> = s.split(&sep).collect();
+                            let len = res.len();
+                            let mut builder = StringColumnBuilder::with_capacity(len, len);
+                            for i in res {
+                                builder.put_slice(i.as_bytes());
+                                builder.commit_row();
+                            }
+                            let column = builder.build();
+                            output.builder.append_column(&column);
```

Comment:
this is inefficient.   You can use the inner builder and manage the offset manually.